itest/starboard_cli_test.go

```diff
@@ -329,6 +332,112 @@ var _ = Describe("Starboard CLI", func() {
 
 		})
 
+		// TODO Run with other integration tests
+		// The only reason this test is marked as pending is that I don't know
+		// how to pass DockerHub private repository credentials to this test case.
+		PContext("when unmanaged Pod with private image is specified as workload", func() {
+			var pod *corev1.Pod
+
+			var podName = "nginx-with-private-image"
+			var secretName = "registry-credentials"
+			var podNamespace = corev1.NamespaceDefault
+
+			BeforeEach(func() {
+				var err error
+				var secret *corev1.Secret
+				secret, err = secrets.NewImagePullSecret(metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: podNamespace,
+				}, "https://index.docker.io/v1",
+					os.Getenv("STARBOARD_TEST_DOCKERHUB_REGISTRY_USERNAME"),
+					os.Getenv("STARBOARD_TEST_DOCKERHUB_REGISTRY_PASSWORD"))
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = kubernetesClientset.CoreV1().Secrets(podNamespace).
+					Create(context.TODO(), secret, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				pod, err = kubernetesClientset.CoreV1().Pods(podNamespace).
+					Create(context.TODO(), &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      podName,
+							Namespace: podNamespace,
+						},
+						Spec: corev1.PodSpec{
+							ImagePullSecrets: []corev1.LocalObjectReference{
+								{
+									Name: secretName,
+								},
+							},
+							Containers: []corev1.Container{
+								{
+									Name:            "nginx",
+									Image:           "starboardcicd/private-nginx:1.16",
```

Comment:
I have registered the `starbaordcicd` user in DockerHub and created the `private-nginx` repository.